cluster-autoscaler/cloudprovider/gce/autoscaling_gke_client.go

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gce
+
+import (
+	"flag"
+	"fmt"
+	"net/http"
+	"time"
+
+	gke "google.golang.org/api/container/v1"
+	gke_beta "google.golang.org/api/container/v1beta1"
+	"k8s.io/autoscaler/cluster-autoscaler/cloudprovider"
+)
+
+// TODO(krzysztof-jastrzebski): Move to main.go.
+var (
+	gkeAPIEndpoint = flag.String("gke-api-endpoint", "", "GKE API endpoint address. This flag is used by developers only. Users shouldn't change this flag.")
+)
+
+const (
+	clusterPathPrefix   = "projects/%s/locations/%s/clusters/%s"
+	nodePoolsPathPrefix = "projects/%s/locations/%s/clusters/%s/nodePools/"
+	nodePoolPathPrefix  = "projects/%s/locations/%s/clusters/%s/nodePools/%%s"
+	operationPathPrefix = "projects/%s/locations/%s/operations/%%s"
+)
+
+// AutoscalingGkeClient is used for communicating with GKE API.
+type AutoscalingGkeClient interface {
+	// reading cluster state
+	FetchLocations() ([]string, error)
+	FetchNodePools() ([]NodePool, error)
+	FetchResourceLimits() (*cloudprovider.ResourceLimiter, error)
+
+	// modifying cluster state
+	DeleteNodePool(string) error
+	CreateNodePool(*Mig) error
+}
+
+// NodePool contains node pool's fields we want to use.
+type NodePool struct {
+	Name              string
+	InstanceGroupUrls []string
+	Autoscaled        bool
+	MinNodeCount      int64
+	MaxNodeCount      int64
+	Autoprovisioned   bool
+}
+
+type autoscalingGkeClientV1 struct {
```

Comment:
I don't feel super-strongly about it, but I would put this in autoscaling_gke_client_v1.go (similarly for v1beta1 client).

Reply:
Done